notes/error.md

```diff
@@ -1,349 +1,444 @@
+However, we would intuitively consider case 2 more accurate than case 1
+since our approximation is double the true value in case 1.
+Because of this, we define the relative error, which will be an error estimate
+independent of the magnitude. To obtain this we simply divide the absolute
+error by the absolute value of the true value.
+
+$$
+\begin{equation}
+\text{Relative Error} = \frac{|x - \hat{x}|}{|x|}
+\end{equation}
+$$
+
+If we consider the two cases again, we can see that the relative error will be much lower in the second case, since relative error is independent of magnitude.
+
+<table class="table">
+<colgroup>
+<col width="50%" />
+<col width="50%" />
+</colgroup>
+<thead>
+<tr class="header">
+<th style="color: black;">Case 1</th>
+<th style="color: black;">Case 2</th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td><span class="math inline">\(x = 0.1\)</span>, <span class="math inline">\(\hat{x} = 0.2\)</span></td>
+<td><span class="math inline">\(x = 100.0\)</span>, <span class="math inline">\(\hat{x} = 100.1\)</span></td>
+</tr>
+<tr class="even">
+<td><span class="math inline">\(\frac{\mid x - \hat{x} \mid}{\mid x \mid} = 1\)</span></td>
+<td><span class="math inline">\(\frac{\mid x - \hat{x} \mid}{\mid x \mid} = 10^{-3}\)</span></td>
+</tr>
+</tbody>
+</table>
+## Significant Digits/Figures
+
+**Significant figures** of a number are digits that carry meaningful information. They are digits beginning from the leftmost nonzero digit and ending with the rightmost "correct" digit, including final zeros that are exact. For example:
+
+*   The number 3.14159 has six significant digits.
+*   The number 0.00035 has two significant digits.
+*   The number 0.000350 has three significant digits.
+<br/>
+The number of significant figures tells us about how many positions of $$x$$ and $$\hat{x}$$ agree.
+
+An approximate result $$\hat{x}$$ has $$n$$ **significant figures** of a true value $$x$$ if the absolute error, $$\vert x - \hat{x}\vert$$, has zeros in the first $$n$$ decimal places counting from the leftmost nonzero (leading) digit of $$x$$, followed by a digit from 0 to 4.
+
+
+**Example:** Assume $$x = 3.141592653$$ and suppose $$\hat{x}$$ is the approximate result:
+
+$$
+\hat{x} = 3.14159 \longrightarrow |x - \hat{x}| = 0.00000\mathbf{2}653 \longrightarrow \hat{x} \text{ has 6 significant figures.}
+$$
+
+$$
+\hat{x} = 3.1415 \longrightarrow |x - \hat{x}| = 0.0000\mathbf{9}2653 \longrightarrow \hat{x} \text{ has 4 significant figures.}
+$$
+
+The number of accurate significant digits can be estimated by the relative error. If
+
+$$
+\text{Relative Error} = \frac{|x - \hat{x}|}{|x|} \geq 10^{-n + 1}
+$$
+
+then $$\hat{x}$$ has **at most** $$n$$ accurate significant digits. 
+
+In general, we will use the rule-of-thumb for calculating an upper bound of the relative error: if an approximation has $$n$$ accurate significant digits, then the relative error is
+
+$$
+\frac{|x - \hat{x}|}{|x|} \leq 10^{-n+1}
+$$
+

```

Comment:
does the rule of thumb need any explanation/intuition/derivation? we use it a lot on hws